src/history/HistoryArchiveReportWork.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "history/HistoryArchiveReportWork.h"
+#include "historywork/GetHistoryArchiveStateWork.h"
+#include "util/Logging.h"
+#include <Tracy.hpp>
+#include <fmt/format.h>
+
+namespace stellar
+{
+HistoryArchiveReportWork::HistoryArchiveReportWork(
+    Application& app,
+    std::vector<std::shared_ptr<HistoryArchive>> const& archives)
+    : Work(app, "history-archive-report-work", BasicWork::RETRY_NEVER)
+    , mGetHistoryArchiveStateWorks()
+    , mAdded(false)
+    , mArchives(archives)
+{
+}
+
+BasicWork::State
+HistoryArchiveReportWork::doWork()
+{
+    ZoneScoped;
+
+    if (!mAdded)
```

Comment:
No need for member variable `mAdded`, you can simply check if `mGetHistoryArchiveStateWorks` is empty.

Reply:
Ah, yes! Removed.